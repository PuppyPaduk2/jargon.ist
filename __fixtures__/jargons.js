// @flow

export default [
  {
    html: '<p>Hello world</p>',
    title: 'Example Jargon 1',
    slug: 'example-jargon-1',
    tags: [{ title: 'tag1', slug: 'tag1' }, { title: 'tag2', slug: 'tag2' }],
    sameWith: ['sameWith1', 'sameWith2'],
  },

  {
    html: '<p>Lorem ipsum dolor sit amet</p>',
    title: 'Example Jargon 2',
    slug: 'example-jargon-2',
    tags: [{ title: 'tag1', slug: 'tag1' }, { title: 'tag2', slug: 'tag2' }],
    sameWith: ['sameWith1', 'sameWith2'],
  },
];
