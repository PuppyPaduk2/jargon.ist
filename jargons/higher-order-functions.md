---
title: Higher Order Functions
tags:
- fonksiyonel programlama
---

`higher order functions` hebele hubele
