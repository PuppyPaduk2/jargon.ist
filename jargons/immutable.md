---
title: immutable
sameWith:
- immutability
tags:
- javascript
- haskell
- ocaml
- fonksiyonel programlama
---

Bu terim uygulamanın herhangi bir noktasında [mutate](/mutate) edilemeyen veya edilmemesi gereken veri tipleri için kullanılır.

Örnek olarak JavaScript için `const x = 5;` tanımlamasındaki `x` değişkeni immutable'dır.
