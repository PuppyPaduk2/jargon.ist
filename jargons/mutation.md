---
title: mutation
sameWith:
- mutate
- mutable
- mutability
- mutate etmek
tags:
- javascript
- haskell
- ocaml
- fonksiyonel programlama
---

Mutation programlama dillerinde ana kaynağı oluşturma aşamasından sonra uygulamanın herhangi bir anında
modifiye etmek anlamına gelir. JavaScript özelinde bakacak olarak herhangi bir array'a bir eleman push'lamak
mutation yaratır. Yine JavaScript özelinde [primitive](/primitive) tipler [immutable](/immutable)
iken `Array` ve `Object` tipleri mutable'dır.

Mutation örneği;

```js
const obj = {
  user: 'user1',
  email: 'user1@example.com',
};

obj.country = 'TR'; // Burada mutation yarattık.
```
