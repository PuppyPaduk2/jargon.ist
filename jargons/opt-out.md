---
title: opt-out
sameWith:
- opt out
tags:
- genel
---

opt-out terimi programlamada bir özelliğin **isteğe bağlı olarak sistemden çıkarılabilmesi** anlamına gelmektedir. Örnek olarak bir log sistemi bir framework'te kapatılabiliyorsa log sistemi bu framework için **opt-out durumdadır**.

Ayrıca tam tersi için [opt-in](/opt-in) terimi incelenebilir.
