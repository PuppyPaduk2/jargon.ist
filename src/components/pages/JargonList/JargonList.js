// @flow

import React, { Fragment } from 'react';
import Link from 'gatsby-link';

import { type Jargon } from '../../../types/jargon';

type Props = {
  jargons: Array<Jargon>,
};

const JargonList = ({ jargons }: Props) => (
  <Fragment>
    <h1>Dizin</h1>
    <ul style={{ margin: 0, padding: 0, listStyle: 'none' }}>
      {jargons.map(jargon => (
        <li key={jargon.slug || ''} className="u-gap-bottom-xsmall">
          <Link to={`/${jargon.slug || ''}`} className="u-text-lowercase">
            {jargon.title}
          </Link>
        </li>
      ))}
    </ul>
  </Fragment>
);

export default JargonList;
