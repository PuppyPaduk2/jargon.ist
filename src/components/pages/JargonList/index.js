// @flow

import JargonList from './JargonList';

export default JargonList;
