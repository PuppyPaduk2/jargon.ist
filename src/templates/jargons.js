// @flow

import React, { Fragment } from 'react';
import Helmet from 'react-helmet';

import JargonList from '../components/pages/JargonList';

import { type Jargon } from '../types/jargon';

type Props = {
  pathContext: {
    jargonList: Array<Jargon>,
  },
};

export const Jargons = ({ pathContext }: Props) => (
  <Fragment>
    <Helmet>
      <title>Dizin</title>
    </Helmet>

    <JargonList jargons={pathContext.jargonList} />
  </Fragment>
);

export default Jargons;
