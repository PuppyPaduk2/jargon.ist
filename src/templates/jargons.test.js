// @flow

import React from 'react';
import { shallow } from 'enzyme';

import { Jargons } from './jargons';

import JargonList from '../components/pages/JargonList';

it('renders correctly', () => {
  const wrapper = shallow(<Jargons pathContext={{ jargonList: [] }} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders title', () => {
  const wrapper = shallow(<Jargons pathContext={{ jargonList: [] }} />);

  expect(wrapper.find('title')).toHaveLength(1);
});

it('renders JargonList component', () => {
  const wrapper = shallow(<Jargons pathContext={{ jargonList: [] }} />);

  expect(wrapper.find(JargonList)).toHaveLength(1);
});
