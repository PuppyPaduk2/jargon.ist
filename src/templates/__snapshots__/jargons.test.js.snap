// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Dizin
    </title>
  </HelmetWrapper>
  <JargonList
    jargons={Array []}
  />
</React.Fragment>
`;
